controllers/capabilities/methods.go

```diff
@@ -107,6 +108,9 @@ func (t *ProductThreescaleReconciler) processNotDesiredMethods(notDesiredMap map
 	for _, notDesiredMethod := range notDesiredMap {
 		err := t.productEntity.DeleteMethod(notDesiredMethod.ID)
 		if err != nil {
+			if strings.Contains(err.Error(), "Method is used by the latest gateway configuration and cannot be deleted") {
+				return fmt.Errorf("Please check Mapping Rules in UI and delete those rules that not in Product CR, or add them to Product CR. Check Mapping Rules in UI that use Method [%s].  %w", notDesiredMethod.SystemName, err)
```

Comment:
This message is not correct Valery.
We do not want to encourage interacting with 3scale assets via CR and UI as mixed. It's either the CR or the UI/API. 
What this should say instead is something like:
```
Method xyz is used by the latest gateway configuration, it will be removed from 3scale only once the promotion without the mapping rule that uses the deleted method is done
```

Reply:
updated. Thank your for new solution